### ISSUE TITLE
Latest version not available at F-Droid

### ISSUE THREAD

**Issue by IzzySoft** [wiglenet/wigle-wifi-wardriving#233]
Looking at the F-Droid repo, I cannot see the last version reported here from `releases/` (i.e. v2.29). Checking with the maintainers, it seems the F-Droid specific branch (`foss-master`) has not been updated.

Any idea what could be done here? First glance says either updating that branch, or telling F-Droid they can pick it from master (which would mean master complies to the F-Droid policy).

Could you please let me know how to proceed with this? Thanks in advance!

**Comment by IzzySoft** [comment]
@bobzilladev thanks a lot! If you close this issue *then* (when the new foss build is available) I can use that as a trigger for the maintainers :wink:

**Comment by bobzilladev** [comment]
Merged in https://github.com/wiglenet/wigle-wifi-wardriving/pull/235

**Comment by bobzilladev** [comment]
Hrm, looks like they have 2.10 on f-droid, the previous foss-master was at 2.25, so looks like they aren't grabbing the latest when we do update the branch. The foss-master branch is now at 2.29 (a little past it since I pulled in master which has some unreleased commits).

**Comment by IzzySoft** [comment]
Yes, I already incited "bad conscience" on the fact that not even 2.25 was built. Reason is mainly "lacking man-power". I did dig this up while walking *the entire repo* checking for inconsistencies (also including source gone, links changed, and currently hunting for incomplete descriptions). Thing is I'm not a dev, so mostly raising issues is the only thing I can do in cases like this. Thankfully another contributor provided the "Build" block (and verified it) – so hopefully all that's left is the MR to be merged, and on the next run fdroidserver should take care for the rest. Keeping my fingers crossed!

**Comment by bobzilladev** [comment]
Nice, thanks again! I think we looked into updating it ourselves at one point, but the process was involved enough to get a local build going that we decided to focus on other priorities. Glad someone else has done that bit :)

**Comment by IzzySoft** [comment]
@bobzilladev sad news, [a maintainer just reported](https://gitlab.com/fdroid/fdroiddata/merge_requests/3414#note_86924813):

> Includes a prebuilt binary: wiglewifiwardriving/libs/TTS_library_stub.jar.

Can you fix that?

**Comment by bobzilladev** [comment]
Interesting, I would guess so, that was for super old android, our current minimum version pry doesn't need that. Will look into.

**Comment by IzzySoft** [comment]
Thanks a lot, @bobzilladev! Just an organizational question: will that result in a new version (230), or do you rather plan to "move the tag" so it stays at 2.29,229? In the latter case I'd try to keep the current MR open, in the former we might need another one.

**Comment by bobzilladev** [comment]
i'd merge as same version, hopefully before our next actual release.

**Comment by IzzySoft** [comment]
That's great, thanks! Updated the MR accordingly (asking to keep it open until then). Plase just ping me when it's ready, so I can route the ping to the maintainers :ping_pong: :round_pushpin:

**Comment by bobzilladev** [comment]
Latest foss-master removes tts jar (was needed for android 1.5).

**Comment by IzzySoft** [comment]
@bobzilladev sorry to annoy you once more: Could you add a "foss" tag whenever you've updated the app? Could e.g. look like `foss-%v` (with `%v` being the version). That would enable us to "detect" new releases. Thanks!

**Comment by bobzilladev** [comment]
Ah, sure. We have a big release staged, I'll wait until that goes out to do the merge deconflict dance, will tag then.

**Comment by bobzilladev** [comment]
We have a new 2.43 release coming out, I've merged and deconflicted the foss-master branch and tagged it foss-2.43.

**Comment by IzzySoft** [comment]
Thanks for the notification! I see AutoUpdateMode is still on "None", so this again will require manual intervention. If you have a GitLab account, you could simply open an issue in fdroiddata whenever a new release should be packaged (or we could enable automated update checking, setting AutoUpdateMode to "Version foss-%v" (assuming versionName is "2.43"). I could setup an MR for the latter if you wish.

What shall it be?

**Comment by bobzilladev** [comment]
We don't touch or maintain f-droid at all. We can make sure to do the foss tag going forward so automated building would work.

**Comment by IzzySoft** [comment]
Sounds like a good plan. @rudloff do you think we could enable AUM on our end then, as I suggested in my previous comment (2 posts up from here)? Then I'd setup the corresponding MR.

**Comment by Rudloff** [comment]
If we now have tags for the foss branch, I suppose we could enable auto update yes (with the proper regexp to only get relevant tags).

**Comment by IzzySoft** [comment]
@Rudloff `AutoUpdateCheck: Version foss-%v` would do if checked against the Manifest's `android:versionName`. The gradle unfortunately lacks both `versionName` and `versionCode`.

**Comment by bobzilladev** [comment]
Feel free to open a PR against the foss-master branch with whatever stringly type matching is required for that build system.

**Comment by IzzySoft** [comment]
I can't (I'm not a dev so I wouldn't know what depends on what else and thus might break things).

**Comment by bobzilladev** [comment]
Looks straightforward to move versioning from manifest to gradle, I can look at that sometime this week.

**Comment by Rudloff** [comment]
I'm not sure you need to change anything.
`fdroid checkupdates` seems to find everything it needs:
```DEBUG: Check tag: 'foss-2.43'
DEBUG: Directory: build/net.wigle.wigleandroid
DEBUG: > git rev-parse --show-toplevel
DEBUG: Directory: build/net.wigle.wigleandroid
DEBUG: > git submodule foreach --recursive git reset --hard
DEBUG: Directory: build/net.wigle.wigleandroid
DEBUG: > git submodule foreach --recursive git clean -dffx
DEBUG: Directory: build/net.wigle.wigleandroid
DEBUG: > git checkout -f foss-2.43
DEBUG: Directory: build/net.wigle.wigleandroid
DEBUG: > git clean -dffx
DEBUG: Parsing manifest at 'build/net.wigle.wigleandroid/build.gradle'
DEBUG: ..got package=None, version=None, vercode=None
DEBUG: Parsing manifest at 'build/net.wigle.wigleandroid/wiglewifiwardriving/src/main/AndroidManifest.xml'
DEBUG: ..got package=net.wigle.wigleandroid, version=2.43, vercode=243
DEBUG: Parsing manifest at 'build/net.wigle.wigleandroid/wiglewifiwardriving/build.gradle'
DEBUG: ..got package=net.wigle.wigleandroid, version=None, vercode=None
DEBUG: Adding possible subdir wiglewifiwardriving
DEBUG: Parsing manifest at 'build/net.wigle.wigleandroid/wiglewifiwardriving/src/main/AndroidManifest.xml'
DEBUG: ..got package=net.wigle.wigleandroid, version=2.43, vercode=243
DEBUG: Parsing manifest at 'build/net.wigle.wigleandroid/wiglewifiwardriving/build.gradle'
DEBUG: ..got package=net.wigle.wigleandroid, version=None, vercode=None
DEBUG: Manifest exists in subdir 'wiglewifiwardriving'. Found version 2.43 (243)
DEBUG: Parsing manifest at 'build/net.wigle.wigleandroid/wiglewifiwardriving/src/main/AndroidManifest.xml'
DEBUG: ..got package=net.wigle.wigleandroid, version=2.43, vercode=243
DEBUG: Adding possible subdir wiglewifiwardriving/src/main
DEBUG: Parsing manifest at 'build/net.wigle.wigleandroid/wiglewifiwardriving/src/main/AndroidManifest.xml'
DEBUG: ..got package=net.wigle.wigleandroid, version=2.43, vercode=243
DEBUG: Manifest exists in subdir 'wiglewifiwardriving/src/main'. Found version 2.43 (243)
```

**Comment by bobzilladev** [comment]
merged new release and tagged as `foss-2.44`

**Comment by drzraf** [comment]
Still not in f-droid.

**Comment by IzzySoft** [comment]
The updater found it, but due to `AutoUpdateMode: None` no build was created. @Rudloff should we enable that, by changing the updater settings to

```yaml
AutoUpdateMode: version foss-%v
UpdateCheckMode: Tags
```

? Though that would require @bobzilladev to stick to that naming convention (there is no `foss-*` tag after 2.44, all newer ones up to 2.48 have no prefix). Maybe until that is clarified, you could create the 2.44 build manually, Pierre?

**Comment by bobzilladev** [comment]
I'll stick to the tagging convention, we just haven't gone through the merge-deconflict for every version.

**Comment by IzzySoft** [comment]
Great! I was guessing that, but there were quite a few new tags so I wasn't sure. Still, I'd rather leave the decision of turning auto-updates on to @Rudloff (or any other packager), as we should do so only after having established a corresponding build the updater can copy the recipe etc. from – which would be a one-time manual job.

**Comment by bobzilladev** [comment]
Pushed a foss-2.49 tag on the foss-master branch. This has the latest (currently beta) code.

**Comment by IzzySoft** [comment]
Thanks Bob! @Rudloff could you give it a try, please?

**Comment by Rudloff** [comment]
I tried to build 2.49 but it uses an untrusted Gradle repository:
```
ERROR: Found unknown maven repo 'https://dl.bintray.com/alt236/maven' at build.gradle
```

**Comment by bobzilladev** [comment]
That houses a Bluetooth LE library we use: https://github.com/alt236/Bluetooth-LE-Library---Android

**Comment by Rudloff** [comment]
We could probably get it from Jitpack (which is one of the few repositories we allow): https://jitpack.io/com/github/alt236/Bluetooth-LE-Library---Android/v1.0.0/

**Comment by rksh** [comment]
honestly, the library isn't doing much in the current implementation - we should do a code review and determine whether we can either 1. use it more fully or 2. dump it in favor of core BTLE features introduced since its creation.

**Comment by bobzilladev** [comment]
rksh removed the library from the mainline, I've merged into foss-master with tag 'foss-2.49.1'.

**Comment by bobzilladev** [comment]
tagged foss-2.50

**Comment by Rudloff** [comment]
I can still see the untrusted repository in `foss-2.50`: https://github.com/wiglenet/wigle-wifi-wardriving/blob/a3eeeba30c309d1d6daa5271f768349da6815586/build.gradle#L10

**Comment by bobzilladev** [comment]
Looks like the reference wasn't removed, that repo isn't used to build anymore. I've removed in master/foss-master and tagged `foss-2.51`.

**Comment by Rudloff** [comment]
Thanks, I added 2.51: https://gitlab.com/fdroid/fdroiddata/-/merge_requests/6317

**Comment by rksh** [comment]
so, what's the "right" way for us to keep this up to date, make everyone happy, and close this ticket?

**Comment by IzzySoft** [comment]
Currently, auto-update seems to be disabled. Last time, Pierre manually added 2.51. MaintainerNotes say "Not every release does make it into foss-master" – looks like that's the reason it's still set to manual.

Looking at the latest build section and comparing that with your tags, it looks like about once a year you indeed create a tag by the pattern `foss-<versionName>`. If that is true, it might be possible to enable auto-updates with that pattern:

```yaml
AutoUpdateMode: Version foss-%v
UpdateCheckMode: Tags
```

Can you confirm this, @rksh? And if so, how often does the "build recipe" change? Looking at the last 5 build blocks, each looks different, which makes auto-updates impossible (they'd simply fail). Which means whenever a new FOSS version is available, someone needs to manually add it. Not optimal.

@Rudloff could you check if `foss-2.52` could be added for now?

**Comment by rksh** [comment]
happy to agree to the naming convention - that's no problem for us at all.
when you say "build blocks" - build gradle files/dependencies? Android and our library list set quite a pace for library version updates, so that get revised rather frequently.

**Comment by IzzySoft** [comment]
when I say "build blocks", I'm referring to your YAML metadata:

```yaml
  - versionName: '2.10'
    versionCode: 210
    commit: 63daa022c4d7cbc43f6d0ac287c3addd077b9bd2
    subdir: wiglewifiwardriving
    gradle:
      - yes
    rm:
      - wiglewifiwardriving/libs/slf4j-android*.jar
    prebuild: sed -i -e '/support-v4/acompile "org.slf4j:slf4j-android:1.7.19"' -e
        '/slf4j-android-1.5.8.jar/d' -e 's/net.wigle.wigleandroid.foss/net.wigle.wigleandroid/g'
        build.gradle
    
  - versionName: '2.29'
    versionCode: 229
    commit: 4d9b02f6f516818458f88fe153613689495e204d
    subdir: wiglewifiwardriving
    gradle:
      - yes
    rm:
      - wiglewifiwardriving/libs/TTS_library_stub.jar
    prebuild: sed -i -e '/TTS_library_stub.jar/d' build.gradle

  - versionName: '2.51'
    versionCode: 251
    commit: foss-2.51
    gradle:
      - yes
    output: wiglewifiwardriving/build/outputs/apk/release/wiglewifiwardriving-release-unsigned.apk
    prebuild: sed -i -e '/TTS_library_stub.jar/d' build.gradle
```

Auto update would take the latest build block to create the new one for. So currently, it would construct

```yaml
  - versionName: '2.52'
    versionCode: 252
    commit: foss-2.52
    gradle:
      - yes
    output: wiglewifiwardriving/build/outputs/apk/release/wiglewifiwardriving-release-unsigned.apk
    prebuild: sed -i -e '/TTS_library_stub.jar/d' build.gradle
```

based on 2.51. [See here](https://gitlab.com/fdroid/fdroiddata/-/blob/master/metadata/net.wigle.wigleandroid.yml) for the full metadata file of Wigle. "Build blocks" is what we also call "build recipes". Usually they are pretty constant, just for Wigle they are permanently changing it seems – which makes auto-updates hard. I'm no Android developer, so I don't have "deep insight" into this – hence apologies if I cannot give detailed help.

**Comment by Rudloff** [comment]
I tried to build foss-2.52 but our scanner returns this:
```
2021-03-13 12:32:56,085 ERROR: Found gzip file archive at etc/wireshark_manuf.gz
2021-03-13 12:32:56,086 ERROR: Found gzip file archive at etc/oui36.csv.gz
2021-03-13 12:32:56,086 ERROR: Found gzip file archive at etc/mam.csv.gz
2021-03-13 12:32:56,086 ERROR: Found gzip file archive at etc/oui.csv.gz
```

Can you please explain what these file contain? Should we remove or whitelist them?

**Comment by bobzilladev** [comment]
We've just pushed updated (version 2.80) code to the foss-main branch, including a signed apk:
https://github.com/wiglenet/wigle-wifi-wardriving/tree/foss-main

Closing out this old ticket.

**Comment by IzzySoft** [comment]
Thanks @bobzilladev – not sure what you mean by "including a signed apk" (I assume you've put one inside the repo; the better place for the APKs would be the [releases](https://github.com/wiglenet/wigle-wifi-wardriving/releases)), but looking at the YAML it seems the last foss tag (2.63) was picked up fine, so this one should be found then, too. If not (it will take a few days for the resulting version to show up at F-Droid, as a fresh [build cycle](https://monitor.f-droid.org/builds/running) had just started earlier today) give us a ping and we can investigate.

**Comment by bobzilladev** [comment]
Aye, we always push a signed apk into the commit on github so people can grab and run.

**Comment by IzzySoft** [comment]
Why not rather attaching that APK to releases? As part of a commit, it just "bloats" the repo unnecessarily (each potential contributor has to check out all those APKs on cloning). AFAIK there's even some Github workflow to perform that task automatically (but I'm neither an Android dev nor do I use Github CI, so I have no details on that).

**Comment by bobzilladev** [comment]
I don't think that github releases existed when this project was started, so the files are committed. Process changes require time and effort, and this isn't a priority.

**Comment by IzzySoft** [comment]
Was just a suggestion :wink: And Github releases exist for a long time now (as long as I can think back).

**Comment by bobzilladev** [comment]
You're not wrong, that would be the right way to do it, it's an artifact not code ☑️. Just want to set expectations that no one is going to get around to changing this and the things that depend on it cuz it's good enough and there's other things we'd rather do 😄

**Comment by IzzySoft** [comment]
Yes, that's clear of course – and I did not expect you to have it changed before dinner :rofl: But now that you're aware of this, it might come in handy "one day". As long as F-Droid is providing the APKs via its repo, noone will complain (APKs can even be downloaded from there directly without the client). Releases is just the first place one would look for otherwise.

**TL;DR:** timing is yours, idea was communicated – no pressure and so on. Keep up the good work!
